lib/Async/Registry/thread_registry.cpp

```diff
@@ -0,0 +1,103 @@
+#include "thread_registry.h"
+
+#include "Assertions/ProdAssert.h"
+#include "Async/Registry/Metrics.h"
+#include "Inspection/Format.h"
+#include "Logger/LogMacros.h"
+#include "Logger/Logger.h"
+#include "Logger/LoggerStream.h"
+#include "Metrics/Counter.h"
+
+using namespace arangodb::async_registry;
+
+auto ThreadRegistry::make(std::shared_ptr<const Metrics> metrics)
+    -> std::shared_ptr<ThreadRegistry> {
+  return std::shared_ptr<ThreadRegistry>(new ThreadRegistry{metrics});
```

Comment:
Unfortunately this is not working because I made the ThreadRegistry constructor private and `std::make_shared` cannot access private methods. I want to make sure that Thread Registry is only used inside a shared pointer because in line 37 we use `shared_from_this()` which will fail if this is not inside a shared ptr.